pulp_gem/specs.py

```diff
@@ -60,8 +203,30 @@ def analyse_gem(file_obj):
     """
     with TarFile(fileobj=file_obj) as archive:
         with archive.extractfile("metadata.gz") as md_file:
-            data = yaml.safe_load(gzip.decompress(md_file.read()))
+            data = yaml.load(gzip.decompress(md_file.read()), Loader=RubyMarshalYamlLoader)
+    gem_info = {
+        "name": data.values["name"],
+        "version": data.values["version"].values["version"],
+    }
+    # Sanitize name
+    if not NAME_REGEX.match(gem_info["name"]):
+        raise ValueError(f"Invalid gem name: {gem_info['name']}")
+    # Sanitize version
+    if VERSION_REGEX.match(gem_info["version"]):
+        gem_info["prerelease"] = False
+    elif PRERELEASE_VERSION_REGEX.match(gem_info["version"]):
+        gem_info["prerelease"] = True
+    else:
+        raise ValueError(f"Invalid version string: {gem_info['version']}")
+    for key in ("required_ruby_version", "required_rubygems_version"):
+        if (requirement := data.values.get(key)) is not None:
+            gem_info[key] = _collapse_requirement(requirement)
+    if (dependencies := data.values.get("dependencies")) is not None:
+        gem_info["dependencies"] = {
+            dep.values["name"]: _collapse_requirement(dep.values["requirement"])
+            for dep in dependencies
+        }
     # Workaroud
     del data.values["date"]
```

Comment:
Curious, do you know why we must delete the 'date' value? Also, can this error if 'date' isn't in the returned values?

Reply:
I think the yaml serializer did not perceive it well. But no, i don't remember.